dipy/reconst/fwdti.py

```diff
@@ -121,16 +123,24 @@ def __init__(self, gtab, fit_method="NLS", *args, **kwargs):
 
         if not callable(fit_method):
             try:
-                fit_method = common_fit_methods[fit_method]
+                self.fit_method = common_fit_methods[fit_method]
             except KeyError as e:
                 e_s = '"' + str(fit_method) + '" is not a known fit '
                 e_s += "method, the fit method should either be a "
                 e_s += "function or one of the common fit methods"
                 raise ValueError(e_s) from e
-        self.fit_method = fit_method
+        else:
+            self.fit_method = fit_method
+
         self.design_matrix = design_matrix(self.gtab)
         self.args = args
-        self.kwargs = kwargs
+        self.kwargs = kwargs.copy()
+
+        # Ensure Diso is set correctly and not overwritten by fit_method
+        self.kwargs["Diso"] = self.kwargs.get("Diso", 3e-3)
+
+        # Remove piterations if it's present (it's not used in nls_iter)
+        self.kwargs.pop("piterations", None)
```

Comment:
can you explain ?

Reply:
You are right, line 140 was unnecessary, I am removing it.
But line 143 I put because I am receiving keyword `piterations` to my function `FreeWaterTensorModel` which has init method with `fit_method=NLS` defined, NLS(nls_iter) doesn't have any `piterations` argument. This error is happening from a single test called `test_fwdti_precision` in `test_fwdti.py`, Line 144, 145.

So solution is either I pop it from kwargs in my function or remove piterations in test.
https://github.com/dipy/dipy/blob/6d30ad44c0b9e2202e28e8eab5e2b15fb26260f8/dipy/reconst/tests/test_fwdti.py#L144

I am open to any other workaround.

Reply:
You are right. These were unnecessary. It is removed.